dask_expr/frameexpr.py

```diff
@@ -0,0 +1,1375 @@
+from __future__ import annotations
```

Comment:
I think I'm not a fan of the name `frameexpr.py`.  Maybe `frame.py`, maybe `dataframe/expr.py`?  Thoughts?